docs/communicate-with-javascript.md

```diff
@@ -826,6 +841,91 @@ var Path = require("path");
 var root = Path.dirname("/User/github");
 ```
 
+### Binding to properties inside a module or global
+
+For cases when we need to create bindings for a property within a module or a
+global JavaScript object, Melange provides the `bs.scope` attribute.
+
+For example, if we want to write some bindings for a specific property
+`commands` from [the `vscode`
+package](https://code.visualstudio.com/api/references/vscode-api#commands), we
+can do:
+
+```ocaml
+type param
+external executeCommands : string -> param array -> unit = ""
+  [@@bs.scope "commands"] [@@bs.module "vscode"] [@@bs.splice]
+
+let f a b c = executeCommands "hi" [| a; b; c |]
+```
+
+Which compiles to:
+
+```javascript
+var Vscode = require("vscode");
+
+function f(a, b, c) {
+  Vscode.commands.executeCommands("hi", a, b, c);
+}
+```
+
+The `bs.scope` attribute can take multiple arguments as payload, in case we want
+to reach deeper into the object from the module we are importing.
+
+For example:
+
+```ocaml
+type t
+
+external back : t = "back"
+  [@@bs.module "expo-camera"] [@@bs.scope "Camera", "Constants", "Type"]
+
+let camera_type_back = back
+```
+
+Which generates:
+
+```javascript
+var ExpoCamera = require("expo-camera");
+
+var camera_type_back = ExpoCamera.Camera.Constants.Type.back;
+```
+
+It can also be used in combination with other attributes besides `bs.module`,
+like `bs.val`:
+
+```ocaml
+external imul : int -> int -> int = "imul" [@@bs.val] [@@bs.scope "Math"]
```

Comment:
I realized `bs.scope` is not necessary when using `bs.val` because the external function name can take dots, e.g.

https://github.com/melange-re/documentation-site/blob/7022ed1542e940dde957bbde0eed501811418f76/docs/communicate-with-javascript.md?plain=1#L801

But as it's supported, why not document it 🤷

Reply:
I don't know if we want to rely on that. I could imagine some transformations we would want to do in the future would benefit from having well-defined semantics.

Reply:
I agree. This made me think we should then get rid of the `bs.val` example what was using `"Math.imul"` string on the right side of the external. Did so in https://github.com/melange-re/documentation-site/pull/26/commits/8f8cdda0767d67505679d198e4f2aaec81480909.

Reply:
> But as it's supported, why not document it 🤷

Oh, just read this 😬 Well, lmk if you want to change it back. I still think if it's supported but we don't want people to rely much on it, and there's a "safer" alternative to it, better to leave it undocumented.